config/validator.go

```diff
@@ -89,14 +89,17 @@ func (c *YamlConfig) RegionsEntryLogicalValidator() (valid bool, validationError
 	for regionName, regionConf := range c.Regions {
 		errList = append(errList, c.validateRegionCluster(regionName, regionConf)...)
 	}
-	if len(errList) > 0 {
-		valid = false
-		errorsList := make(map[string][]error)
-		errorsList["RegionsEntryLogicalValidator"] = errList
-		validationErrors = mergeErrors(validationErrors, errorsList)
-	} else {
-		valid = true
+	validationErrors, valid = prepareErrors(errList, "RegionsEntryLogicalValidator")
+	return
+}
+
+// TransportsEntryLogicalValidator checks the correctness of "Transports" part of configuration file
+func (c *YamlConfig) TransportsEntryLogicalValidator() (valid bool, validationErrors map[string][]error) {
+	errList := make([]error, 0)
+	if len(c.Service.Client.Transports) == 0 {
```

Comment:
In case some magic happens, I would suggest to write the condition as follows `if len(c.Service.Client.Transports) < 1`

Reply:
`case some magic happens` - what exactlly ?

Reply:
Idk, it's just a sanity check. Range conditions are imho just "safer" in such cases

Reply:
but in this case `len` function never got minus value and for below case `== 0` in condition is enough